js/modules/k6/ws/ws_test.go

```diff
@@ -504,3 +539,53 @@ func TestTLSConfig(t *testing.T) {
 	})
 	assertSessionMetricsEmitted(t, stats.GetBufferedSamples(samples), "", sr("WSSBIN_URL/ws-close"), 101, "")
 }
+
+func TestReadPump(t *testing.T) {
+	var closeCode int
```

Comment:
I don't like this shared variable

Reply:
Meh, I'm ambivalent about it. It's there so each test can alter the close code returned by the server. Yes, we can't parallelize them this way, but these tests run in milliseconds, so it shouldn't be needed.

I could fix this by making the handler a struct with a `closeCode` field and a `Handle` function or creating different endpoints to handle different close codes, but they both seem overkill for this test.

Let me know if you feel strongly about this, and please propose your preferred way of fixing it.

Reply:
if it runs okay with `go test -race -count 200` I am fine. otherwise I would advice you to use [atomic](https://golang.org/pkg/sync/atomic/)

Reply:
@MStoykov Yes, `go test -v -race -count 200 -run 'TestReadPump' ./js/modules/k6/ws` passes.